designs/2018-programmatic-style-edits/README.md

```diff
@@ -0,0 +1,401 @@
+* **Style enforcer** - any programmatic way to enforce code style. This includes ESLint shareable configs and tools like [Prettier](https://prettier.io/).
+
+### How ESLint Style Enforcers Work Today
+
+Creating a style enforcer in ESLint today generally means the following:
+
+1. Reviewing each of the stylistic ESLint rules and determine the correct settings for each. This process tends to be the same whether end-users create a new style enforcer or are working off of an existing style enforcer that isn't completely to their liking.
+2. Create a shareable config that encapsulates all of the desired ESLint rule settings.
+2. Use `extends` in your project to inherit from that shareable config.
+3. Run ESLint with the style enforcer configuration to see if there are any stylistic warnings, or run ESLint with `--fix` to automatically fix stylistic issues.
+
+In short, the process for style enforcers is the same as for any other type of rule in ESLint (this is, of course, by design). When end-users are content to use an already-existing style enforcer, there is very little burden on the ESLint team as the owners of the individual style enforcers can handle change requests.
+
+The problem is when end-users want to customize a style enforcer to fit their specific needs. There are three basic scenarios that play out every time an end-user wants to customize 
+
+* **They want a change that is already supported by a core rule.** In this case, end-users can use the documentation to determine the the correct rules and rule options to use to get their desired result. This is the case ESLint was optimized for.
+* **They want a change that fits with an existing core rule by no option exists.** In this case, end-users make a request to the ESLint team to enhance an existing rule with new options. The ESLint team then must spend time evaluating the proposal, ensuring consistency of option names across core rules, ensuring no overlap with existing rules, and whether there is someone interested enough to implement the change. 
+* **They want a change that doesn't fit with an existing core rule.** In this case, there are actually two possible outcomes:
+    1. The end-user requests a new core rule be added to ESLint. Once again, this brings with it the burden of evaluating, researching, and potentially implementing a new rule.
+    1. The end-user decides to create a custom rule. Some end-users are happy to do this and distribute their own plugins, however, ESLint limitations such as the inability to easily distribute plugins with shareable configs (https://github.com/eslint/eslint/issues/3458) means that many end-users prefer proposing their own style rules to be included with ESLint. (There is also a bit of halo effect around core rules, where end-users feel that getting their own rules into the ESLint core means they are "blessed" and therefore more valid than custom rules distributed in plugins.)
```

Comment:
That's a good point, there is definitely a parallel problem here. I do believe that style editors solve the problem in a different way than rules because you are not required to name, define options for, or otherwise package up individual configurable options. In fact, I suspect in most cases that style editors will not have many, if any, options, because the point of enforcing a style guide is that you don't want people to have options -- the code should be the One True Way.